// Copyright (C) 2020 THL A29 Limited, a Tencent company.
// All rights reserved.
// Licensed under the BSD 3-Clause License (the "License"); you may
// not use this file except in compliance with the License. You may
// obtain a copy of the License at
// https://opensource.org/licenses/BSD-3-Clause
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" basis,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
// implied. See the License for the specific language governing
// permissions and limitations under the License.
// See the AUTHORS file for names of contributors.

syntax = "proto3";

service CudaAllocator {
  rpc Malloc(MallocRequest) returns (MallocReply) {}
  rpc Free(FreeRequest) returns (FreeReply) {}
}

message MallocRequest { uint64 size = 1; }

message Allocation { uint64 ptr = 1; }

message MallocReply {
  bytes mem_handle = 1;
  Allocation allocation = 2;
}

message FreeRequest { uint64 ptr_to_free = 1; }

message FreeReply {}
